### PR TITLE
Add prepare_lnurl_pay

### LINKED ISSUE #468: Calculate the fee upfront when using LNURL-pay.

### PR DESCRIPTION
**Pull request by dangeross**
This PR adds a `prepare_lnurl_pay()` interface method as a step before `lnurl_pay()` to calculate the fees involved with the payment, whether it is a payment via Lightning or paid directly to a Liquid address using the Magic Route Hint retrieved in the invoice.

Depends on https://github.com/breez/breez-sdk-greenlight/pull/1104 (CI tests will be broken)

Fixes #468

### CODE REVIEW

**Review by ok300 — APPROVED** [review]
Looks good to me.

Validation and fee calculation happens in `prepare_lnurl_pay`, which makes sense.

File: lib/bindings/src/breez_sdk_liquid.udl [hunk 1]
```diff
@@ -344,6 +349,22 @@ dictionary CheckMessageResponse {
...
 
+dictionary PrepareLnUrlPayRequest {
+    LnUrlPayRequestData data;
+    u64 amount_msat;
+    string? comment = null;
+    boolean? validate_success_action_url = null;
+};
+
+dictionary PrepareLnUrlPayResponse {
+    PrepareSendResponse prepare_send_response;
```
Comment (roeierez) [inline lib/bindings/src/breez_sdk_liquid.udl:360]: IMO it is better to have the destination and fee_sat directly under the PrepareLnUrlPayResponse and not use the PrepareSendResponse for that. It reduces decoupling between the two methods.

**Review by roeierez — CHANGES_REQUESTED** [review]
Looks good, one small comment.

File: lib/bindings/src/breez_sdk_liquid.udl (hunk 1 above)
Comment (dangeross) [reply lib/bindings/src/breez_sdk_liquid.udl]: Agreed. Done [27313d5](https://github.com/breez/breez-sdk-liquid/pull/524/commits/27313d5616373424875cfd58396aab99271b8957)

**Review by roeierez — APPROVED** [review]
LGTM

**Pull request merged**